### PR TITLE
Kernel fusion rms norm layer

### PR DESCRIPTION
**Pull request by loopinf**
## Title

Re-arrange fused norm layer and rms layer

## Description

Separate functions and class from autograd  with modules 

## Linked Issues

- resolved #00

### CODE REVIEW

File: oslo/torch/_C/csrc/includes/type_shim.h [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+      x[tid] = final; // EpilogueOp
+    // Make sure the smem result is visible to all warps.
+    __syncthreads();
+  }
+
+  return final;
+}
+
+template <typename T>
+__device__ __forceinline__ T reduce_block_into_lanes_max_op(T *x,
```
Comment (hyunwoongko) [inline oslo/torch/_C/csrc/includes/type_shim.h:331]: This should be restored.

File: oslo/torch/_C/csrc/CUDABinder_ref.cpp [hunk 2]
```diff
@@ -0,0 +1,479 @@
+#include <cuda_fp16.h>
```
Comment (hyunwoongko) [inline oslo/torch/_C/csrc/CUDABinder_ref.cpp:1]: This file should be removed.

File: oslo/torch/nn/__init__.py [hunk 3]
```diff
@@ -3,10 +3,23 @@
...
+    mixed_dtype_fused_rms_norm_affine,
+    fused_rms_norm,
 )
 from oslo.torch.nn.modules.dropout import (
     FusedBiasDropout,
     FusedBiasDropoutResidual,
 )
 from oslo.torch.nn.modules.linear import LazyLinear
 from oslo.torch.nn.modules.conv import Conv1D, LazyConv1D
+
```
Comment (hyunwoongko) [inline oslo/torch/nn/__init__.py:19]: Conflict

File: oslo/torch/_C/csrc/CUDABinder_ref.cpp (hunk 2 above)
Comment (loopinf) [reply oslo/torch/_C/csrc/CUDABinder_ref.cpp]: done removing

File: oslo/torch/_C/csrc/includes/type_shim.h (hunk 1 above)
Comment (loopinf) [reply oslo/torch/_C/csrc/includes/type_shim.h]: Done restoring

File: test_ds.py [hunk 4]
```diff
@@ -0,0 +1,4 @@
+from oslo.torch._C import FusedLayerNormBinder
```
[the remark this answers is not in the document]
Comment (loopinf) [reply test_ds.py]: removed

File: oslo/torch/nn/__init__.py (hunk 3 above)
Comment (loopinf) [reply oslo/torch/nn/__init__.py:25]: Resolve conflict

**Pull request closed without merging**